app/api/relationships.v2/services/propertyUpdateStrategies/UpdateRelationshipPropertiesJob.ts

```
@@ -0,0 +1,31 @@
+import { TransactionManager } from 'api/common.v2/contracts/TransactionManager';
+import { EntityRelationshipsUpdateService } from 'api/entities.v2/services/EntityRelationshipsUpdateService';
+import { Job } from 'api/queue.v2/contracts/Job';
+
+interface IndexEntitiesCallback {
+  (sharedId: string): Promise<void>;
+}
+export class UpdateRelationshipPropertiesJob extends Job {
+  private entityId: string;
+
+  updater?: EntityRelationshipsUpdateService;
```

Things are optional, but then we expect them to be there with '!'. I assume these are supplied by the environment somehow through that registration/definition process. If we keep that as it is, even then I would not just bypass typescript using the exclamation mark, but rather implement proper checks that throw errors when some of this is still undefined when trying to use. I think we did some things that needed initialization after the constructor and before use, and that's how we did it.

---

I will surely implement checks, but they will only add a prettier message. This would also throw an exception in runtime :)

---

I agree. I was more trying to be consistent, rather than fault the usage, out of context it's fine. If we used things (optional-then-exclamation mark) like this habitually, I would not have commented.
And actually, if we think that these kind of errors do not need prettier messages, we can start using this pattern here and leave it as is. I'll leave the decision to you., I'm fine either way.

---

I think maybe the only difference is that this is not user facing, so I was biased towards thinking that not having a better error was ok. But yeah, let's implement proper checks